proactive/ProActiveGateway.py

```diff
@@ -1,39 +1,64 @@
-        gg = JavaGateway.launch_gateway(classpath=os.path.normpath(current_path), die_on_exit=True )
-        
+        self.base_url = base_url        
+        self.gateway = JavaGateway()               
+        self.gg = self.gateway.launch_gateway(classpath=os.path.normpath(current_path), die_on_exit=True )
+  
+
+    def connect(self,username, password, credentials_path=None, insecure=True ):
         credentials_file = None
         if credentials_path is not None:
-            credentials_file = gg.jvm.java.io.File(credentials_path)
-             
+            credentials_file = self.gg.jvm.java.io.File(credentials_path)
             
-        self.proactive_schduler_client = gg.jvm.org.ow2.proactive_grid_cloud_portal.smartproxy.RestSmartProxyImpl()
-        connection_info = gg.jvm.org.ow2.proactive.authentication.ConnectionInfo(base_url + "/rest", username, password, credentials_file, insecure)
+        self.proactive_schduler_client = self.gg.jvm.org.ow2.proactive_grid_cloud_portal.smartproxy.RestSmartProxyImpl()
+        connection_info = self.gg.jvm.org.ow2.proactive.authentication.ConnectionInfo(self.base_url + "/rest", username, password, credentials_file, insecure)
         self.proactive_schduler_client.init(connection_info);
-    
-    def submitFromCatalog(self, bucket_name, workflow_name):
         
-        return self.proactive_schduler_client.submitFromCatalog(self.base_url + "/catalog",bucket_name, workflow_name).getReadableName();
+    def disconnect(self):
+        self.proactive_schduler_client.disconnect()        
+    
+    def submitFromCatalog(self, bucket_name, workflow_name, workflow_variables = {}):
+        workflow_variables_java_map = MapConverter().convert(workflow_variables, self.gg._gateway_client)
+        return self.proactive_schduler_client.submitFromCatalog(self.base_url + "/catalog",bucket_name, workflow_name, workflow_variables_java_map).longValue();
 
+    def submitFile(self, workflow_xml_file_path, workflow_variables = {}):
+        workflow_variables_java_map = MapConverter().convert(workflow_variables, self.gg._gateway_client)
+        return self.proactive_schduler_client.submit(self.gg.jvm.java.io.File(workflow_xml_file_path), workflow_variables_java_map).longValue();
+    
+    def submitURL(self, workflow_url_spec, workflow_variables = {}):
+        workflow_variables_java_map = MapConverter().convert(workflow_variables, self.gg._gateway_client)
+        return self.proactive_schduler_client.submit(self.gg.jvm.java.net.URL(workflow_url_spec), workflow_variables_java_map).longValue();
+    
+    def getJobState(self,job_id):
+        return self.proactive_schduler_client.getJobState(job_id).getName();
     
+    def isJobFinished(self,job_id):
+        return self.proactive_schduler_client.isJobFinished(job_id);
+    
+    def getJobInfo(self,job_id):
+        return self.proactive_schduler_client.getJobInfo(str(job_id));
+    
+    def getAllJobs(self, max_number_of_jobs=1000):
```

Comment:
if you mean to return all running jobs, you should name it getRunningJobs

Reply:
no, I want to return ALL the jobs whatever is the status. It is then up to the user to filter out what he needs

Reply:
ok, but then if you want to return all jobs, you should not do JobFilterCriteria(False, False, True, False), but  JobFilterCriteria(False, True, True, True)